### PR TITLE
DOCS: The interpretation of OPENSSL_API_COMPAT has changed, update docs

### PR DESCRIPTION
**Pull request by levitte**
##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man7/openssl_user_macros.pod.in [hunk 1]
```diff
@@ -23,21 +23,16 @@ user defined macros.
...
 
 =over 4
 
 =item C<0xMNNFF000L>
 
 This is the form supported for all versions up 1.1.x, where C<M>
 represents the major number, C<NN> represents the minor number, and
-C<FF> represents the fix number.  For version 1.1.0, that's
-C<0x10100000L>.
+C<FF> represents the fix number, all as hexadecimal numbers.
```
Comment (richsalz) [inline doc/man7/openssl_user_macros.pod.in]: "as a single hexadecimal number"  (It really is only ONE number, not multiple)

Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in]: Good point

Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in]: Fixed

File: doc/man7/openssl_user_macros.pod.in [hunk 2]
```diff
@@ -23,21 +23,16 @@ user defined macros.
...
 
 =over 4
 
 =item C<0xMNNFF000L>
 
 This is the form supported for all versions up 1.1.x, where C<M>
 represents the major number, C<NN> represents the minor number, and
-C<FF> represents the fix number.  For version 1.1.0, that's
-C<0x10100000L>.
+C<FF> represents the fix number, as a hexadecimal numbers.
```
Comment (InfoHunter) [inline doc/man7/openssl_user_macros.pod.in]: Should this be `as hexadecimal numbers`?

File: doc/man7/openssl_user_macros.pod.in [hunk 3]
```diff
@@ -57,32 +52,39 @@ For convenience, higher numbers are accepted as well, as long as
...
+
+This form is a simple decimal number calculated with this formula:
+
+I<major> * 10000 + I<minor> * 100 + I<patch>
 
-This form is a simple number that represents the major version number
-and is supported for version 3.0 and up.  For extra convenience,
-these numbers are also available:
+where I<major>, I<minor> and I<patch> are the desired major,
+minor and patch numbers of the major number.  For example:
```
Comment (InfoHunter) [inline doc/man7/openssl_user_macros.pod.in]: This reads strange: ...desired major, minor and patch numbers of the *major* number?

File: doc/man7/openssl_user_macros.pod.in (hunk 2 above)
Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in]: Ah, no, "numbers" should be "number". Good catch, though!

File: doc/man7/openssl_user_macros.pod.in (hunk 3 above)
Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in]: Yup. Typo, that "major" should be "version"

**Comment by kaduk** [comment]
Probably want to s/hass/has/ in the commit message...

File: doc/man7/openssl_user_macros.pod.in (hunk 3 above)
Comment (richsalz) [reply doc/man7/openssl_user_macros.pod.in]: Ping @mattcaswell for statement on the Oxford comma (adding comma in two places after minor)

File: doc/man7/openssl_user_macros.pod.in (hunk 2 above)
Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in]: Fixed

File: doc/man7/openssl_user_macros.pod.in (hunk 3 above)
Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in]: Fixed the *major* issue.  The Oxford comma is debatable in my view.

**Review by InfoHunter — APPROVED** [review]
I think the Oxford comma is 'optional'. Approve if it's changed to Oxford comma or not :-)

File: doc/man7/openssl_user_macros.pod.in [hunk 4]
```diff
@@ -23,21 +23,16 @@ user defined macros.
...
-be declared.
-
-The version number assigned to this macro can take one of two forms:
+The value is a version number, given in one of the following two forms:
 
 =over 4
 
 =item C<0xMNNFF000L>
 
 This is the form supported for all versions up 1.1.x, where C<M>
```
Comment (mspncp) [inline doc/man7/openssl_user_macros.pod.in]: all versions up **to** 1.1.x

File: doc/man7/openssl_user_macros.pod.in [hunk 5]
```diff
@@ -57,32 +52,39 @@ For convenience, higher numbers are accepted as well, as long as
...
 C<{- join('', map { my @x = split /=/,$_; $x[1] }
-              grep /^OPENSSL_MIN_API=/, @{$config{openssl_api_defines} // []})
+              grep /^OPENSSL_CONFIGURED_API=/, @{$config{openssl_api_defines} // []})
      || '0x00000000L'
   -}>.
 
+=item B<OPENSSL_NO_DEPRECATED>
+
+If this macro is defined, all deprecated public symbols in all OpenSSL
+versions up to and including the version given with B<OPENSSL_API_COMPAT>
```
Comment (mspncp) [inline doc/man7/openssl_user_macros.pod.in]: given with -> given by

_The former sounds odd to me,  see also https://idioms.thefreedictionary.com/given+with._

**Review by mspncp — COMMENTED** [review]
Two nits.

File: doc/man7/openssl_user_macros.pod.in (hunk 4 above)
Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in]: Fixed

File: doc/man7/openssl_user_macros.pod.in (hunk 5 above)
Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in]: Done

**Review by mspncp — APPROVED** [review]

**Review by InfoHunter — APPROVED** [review]
Approve again

**Comment by levitte** [comment]
Merged.

cc4ff1061efd95f1560289f1bb726975cb7b83fc DOCS: The interpretation of OPENSSL_API_COMPAT has changed, update docs

**Pull request closed without merging**